UltiSnips/django.snippets

```diff
@@ -1,238 +1,367 @@
 endsnippet
 
-snippet %%
-{% ${1:tag_name} %}
-	${2}
-{% end$1 %}
+snippet ffilepath "FilePathField (ffilepath)"
+${1:FIELDNAME} = forms.FilePathField($2)
 endsnippet
 
-snippet {
-{{ ${1} }}${2}
+snippet ffloat "FloatField (ffloat)"
+${1:FIELDNAME} = forms.FloatField($2)
 endsnippet
 
-# Template Tags
+snippet fip "IPAddressField (fip)"
+${1:FIELDNAME} = forms.IPAddressField($2)
+endsnippet
 
-snippet autoescape
-{% autoescape ${1:off} %}
-	${2}
-{% endautoescape %}
+snippet fimg "ImageField (fimg)"
+${1:FIELDNAME} = forms.ImageField($2)
 endsnippet
 
-snippet block
-{% block ${1} %}
-	${2}
-{% endblock $1 %}
+snippet fint "IntegerField (fint)"
+${1:FIELDNAME} = forms.IntegerField($2)
 endsnippet
 
-snippet #
-{# ${1:comment} #}
+snippet fmochoice "ModelChoiceField (fmochoice)"
+${1:FIELDNAME} = forms.ModelChoiceField($2)
 endsnippet
 
-snippet comment
-{% comment %}
-	${1}
-{% endcomment %}
+snippet fmomuchoice "ModelMultipleChoiceField (fmomuchoice)"
+${1:FIELDNAME} = forms.ModelMultipleChoiceField($2)
 endsnippet
 
-snippet cycle
-{% cycle ${1:val1} ${2:val2} ${3:as ${4}} %}
+snippet fmuval "MultiValueField (fmuval)"
+${1:FIELDNAME} = forms.MultiValueField($2)
 endsnippet
 
-snippet debug
-{% debug %}
+snippet fmuchoice "MultipleChoiceField (fmuchoice)"
+${1:FIELDNAME} = forms.MultipleChoiceField($2)
 endsnippet
 
-snippet extends
-{% extends "${1:base.html}" %}
+snippet fnullbool "NullBooleanField (fnullbool)"
+${1:FIELDNAME} = forms.NullBooleanField($2)
 endsnippet
 
-snippet filter
-{% filter ${1} %}
-	${2}
-{% endfilter %}
+snippet freg "RegexField (fregex)"
+${1:FIELDNAME} = forms.RegexField($2)
 endsnippet
 
-snippet firstof
-{% firstof ${1} %}
+snippet fslug "SlugField (fslug)"
+${1:FIELDNAME} = forms.SlugField($2)
 endsnippet
 
-snippet for
-{% for ${1} in ${2} %}
-	${3}
-{% endfor %}
+snippet fsdatetime "SplitDateTimeField (fsdatetime)"
+${1:FIELDNAME} = forms.SplitDateTimeField($2)
 endsnippet
 
-snippet empty
-{% empty %}
-	${1}
+snippet ftime "TimeField (ftime)"
+${1:FIELDNAME} = forms.TimeField($2)
 endsnippet
 
-snippet if
-{% if ${1} %}
-	${2}
-{% endif %}
+snippet ftchoice "TypedChoiceField (ftchoice)"
+${1:FIELDNAME} = forms.TypedChoiceField($2)
 endsnippet
 
-snippet else
-{% else %}
-	${1}
+snippet ftmuchoice "TypedMultipleChoiceField (ftmuchoice)"
+${1:FIELDNAME} = forms.TypedMultipleChoiceField($2)
 endsnippet
 
-snippet ifchanged
-{% ifchanged %}${1}{% endifchanged %}
+snippet furl "URLField (furl)"
+${1:FIELDNAME} = forms.URLField($2)
 endsnippet
 
-snippet ifequal
-{% ifequal ${1} ${2} %}
-	${3}
-{% endifequal %}
+###########################################################################
+#                             MODELS SNIPPETS                             #
+###########################################################################
+
+snippet model "Model"
+class ${1:MODELNAME}(models.Model):
+	${0}
+	class Meta:
+		verbose_name = "$1"
+		verbose_name_plural = "$1s"
+
+	def __str__(self):
```

Comment:
this should return an empty string, otherwise it is not a legal __str__ implementation.